apps/retrospective/services/sprint.go

```diff
@@ -418,30 +443,63 @@ func (service SprintService) SyncSprintData(sprintID string) (err error) {
 		return err
 	}
 
-	tickets, err := tasktracker.GetSprintTaskList(taskProviderConfig, sprint.SprintID)
+	taskTrackerTaskKeySet, err := service.fetchAndUpdateTaskTrackerTask(sprint, taskProviderConfig)
```

Comment:
Add a ToDo to refactor this flow, as discussed.

Reply:
done